presto-main/src/test/java/com/facebook/presto/operator/TestHashJoinOperator.java

```diff
@@ -328,20 +342,22 @@ public void testInnerJoinWithFailingSpill(boolean probeHashEnabled, List<WhenSpi
             default:
                 throw new IllegalArgumentException(format("Unsupported option: %s", whenSpillFails));
         }
-        try {
-            innerJoinWithSpill(probeHashEnabled, whenSpill, buildSpillerFactory, partitioningSpillerFactory);
-            fail("Exception not thrown");
-        }
-        catch (RuntimeException exception) {
-            assertEquals(exception.getMessage(), expectedMessage);
+        for (boolean dictionaryProcessing : ImmutableList.of(true, false)) {
```

Comment:
This should be test method param. Otherwise we're reusing `DummySpillerFactory` which are not intended to be reused. (less test case isolation)